library/std/src/os/unix/net/udp.rs

```diff
@@ -0,0 +1,249 @@
+use super::IpAncillary;
+use crate::{
+    io::{self, IoSlice, IoSliceMut},
+    net::{SocketAddr, ToSocketAddrs, UdpSocket},
+    sys_common::AsInner,
+};
+
+impl UdpSocket {
+    /// Sets the value of the `IPV6_RECVHOPLIMIT` option for this socket.
+    ///
+    /// If enabled, received packets will come with ancillary data ([`IpAncillary`]) providing
+    /// the hop-limit value of the packet.
+    ///
+    /// # Examples
+    ///
+    ///```no_run
+    /// #![feature(unix_socket_ancillary_data)]
+    /// use std::net::UdpSocket;
+    ///
+    /// fn main() -> std::io::Result<()> {
+    ///     let socket = UdpSocket::bind("[::1]:34254")?;
+    ///     socket.set_recvhoplimit(true)?;
+    ///     Ok(())
+    /// }
+    /// ```
+    #[unstable(feature = "unix_socket_ancillary_data", issue = "76915")]
+    pub fn set_recvhoplimit(&self, recvhoplimit: bool) -> io::Result<()> {
+        self.as_inner().set_recvhoplimit(recvhoplimit)
+    }
+
+    /// Get the current value of the socket for receiving TTL in [`IpAncillary`].
+    /// This value can be change by [`set_recvhoplimit`].
+    ///
+    /// Get the socket option `IPV6_RECVHOPLIMIT`.
+    ///
+    /// [`set_recvhoplimit`]: UdpSocket::set_recvhoplimit
+    #[unstable(feature = "unix_socket_ancillary_data", issue = "76915")]
+    pub fn recvhoplimit(&self) -> io::Result<bool> {
+        self.as_inner().recvhoplimit()
+    }
+
+    /// Sets the value of the `IP_RECVTTL` option for this socket.
+    ///
+    /// If enabled, received packets will come with ancillary data ([`IpAncillary`]) providing
+    /// the time-to-live (TTL) value of the packet.
+    ///
+    /// # Examples
+    ///
+    ///```no_run
+    /// #![feature(unix_socket_ancillary_data)]
+    /// use std::net::UdpSocket;
+    ///
+    /// fn main() -> std::io::Result<()> {
+    ///     let socket = UdpSocket::bind("127.0.0.1:34254").expect("couldn't bind to address");
+    ///     socket.set_recvttl(true).expect("set_recvttl function failed");
+    ///     Ok(())
+    /// }
+    /// ```
+    #[unstable(feature = "unix_socket_ancillary_data", issue = "76915")]
+    pub fn set_recvttl(&self, recvttl: bool) -> io::Result<()> {
+        self.as_inner().set_recvttl(recvttl)
+    }
+
+    /// Get the current value of the socket for receiving TTL in [`IpAncillary`].
+    /// This value can be change by [`set_recvttl`].
+    ///
+    /// Get the socket option `IP_RECVTTL`.
+    ///
+    /// [`set_recvttl`]: UdpSocket::set_recvttl
+    #[unstable(feature = "unix_socket_ancillary_data", issue = "76915")]
+    pub fn recvttl(&self) -> io::Result<bool> {
+        self.as_inner().recvttl()
+    }
+
+    /// Receives data and ancillary data from socket.
```

Comment:
*ancilliary data* should link to `IpAncillary`'s docs.

Reply:
How do I add this link?

Reply:
Check the `Ipv4Addr` doc comments for an example.